src/platform/api.ts

```diff
@@ -49,6 +51,14 @@ export interface IExtensionApi {
      * There are a specific set of extensions that are currently allowed to access this API.
      */
     getKernelService(): Promise<IExportedKernelService | undefined>;
+    /**
+     * Returns the suggested controller for a give Jupyter server and notebook.
+     */
+    getSuggestedController(
+        id: string,
```

Comment:
What's id in this case? Kernel id?

Reply:
Oh no it's actually the id of the IJupyterRemoteServerUriProvider. Maybe put a comment here? Or rename the parameter?